### PR TITLE
Label all entities with deploy.sh and ansible plyabooks

### PR DESCRIPTION
**Pull request by dmvolod**

### CODE REVIEW

File: templates/install/ansible/roles/address_controller_multitenant/tasks/main.yml [hunk 1]
```diff
@@ -6,8 +6,14 @@
 - name: Create cluster wide roles used by enmasse-admin service account
   shell: oc apply -n {{ namespace }} -f {{ playbook_dir }}/resources/cluster-roles/address-space-controller.yaml
 - name: Grant address space controller cluster role privileges to service account
-  shell: oc adm policy add-cluster-role-to-user enmasse.io:address-space-controller -n {{ namespace }} system:serviceaccount:{{ namespace }}:enmasse-admin
+  shell: oc adm policy add-cluster-role-to-user enmasse.io:address-space-controller -n {{ namespace }} system:serviceaccount:{{ namespace }}:enmasse-admin --rolebinding-name={{ namespace }}-address-space-controller-enamsse-admin
+- name: Label clusterrolebindings {{ namespace }}-address-space-controller-enamsse-admin
```
Comment (kornys) [inline templates/install/ansible/roles/address_controller_multitenant/tasks/main.yml]: typo enamsse should be enmasse

File: templates/install/ansible/roles/address_controller_multitenant/tasks/main.yml [hunk 2]
```diff
@@ -6,8 +6,14 @@
...
+  shell: oc label --overwrite -n {{ namespace }} clusterrolebindings {{ namespace }}-address-space-controller-enamsse-admin app=enmasse
```
Comment (kornys) [inline templates/install/ansible/roles/address_controller_multitenant/tasks/main.yml]: same typo

File: templates/install/ansible/roles/service_catalog/tasks/main.yml [hunk 3]
```diff
@@ -9,9 +9,13 @@
 - name: Create cluster wide roles used by service broker
   shell: oc apply -n {{ namespace }} -f {{ playbook_dir }}/resources/cluster-roles/service-broker.yaml
 - name: Grant service-broker cluster role privileges to service account
-  shell: oc adm policy add-cluster-role-to-user enmasse.io:service-broker -n {{ namespace }} system:serviceaccount:{{ namespace }}:enmasse-admin
+  shell: oc adm policy add-cluster-role-to-user enmasse.io:service-broker -n {{ namespace }} system:serviceaccount:{{ namespace }}:enmasse-admin --rolebinding-name={{ namespace }}-service-broker-enamsse-admin
```
Comment (kornys) [inline templates/install/ansible/roles/service_catalog/tasks/main.yml]: same typo

File: templates/install/ansible/roles/service_catalog/tasks/main.yml [hunk 4]
```diff
@@ -9,9 +9,13 @@
...
+- name: Label clusterrolebindings {{ namespace }}-service-broker-enamsse-admin
```
Comment (kornys) [inline templates/install/ansible/roles/service_catalog/tasks/main.yml]: same typo

File: templates/install/ansible/roles/service_catalog/tasks/main.yml [hunk 5]
```diff
@@ -9,9 +9,13 @@
...
+  shell: oc label --overwrite -n {{ namespace }} clusterrolebindings {{ namespace }}-service-broker-enamsse-admin app=enmasse
```
Comment (kornys) [inline templates/install/ansible/roles/service_catalog/tasks/main.yml]: same typo

File: templates/install/ansible/roles/standard_authservice/tasks/main.yml [hunk 6]
```diff
@@ -12,8 +12,12 @@
 - name: Create secret with the keycloak credentials
   when: secret_exists.failed
   shell: oc create secret generic -n {{ namespace }} keycloak-credentials --from-literal=admin.username=admin --from-literal=admin.password={{ keycloak_admin_password }}
+- name: Label secret keycloak-credentials
+  shell: oc label --overwrite -n {{ namespace }} secret keycloak-credentials app=enmasse
 - name: Give user lookup permissions for the keycloak controller
-  shell: oc adm policy add-cluster-role-to-user enmasse.io:keycloak-controller system:serviceaccount:{{ namespace }}:enmasse-admin
+  shell: oc adm policy add-cluster-role-to-user enmasse.io:keycloak-controller system:serviceaccount:{{ namespace }}:enmasse-admin --rolebinding-name={{ namespace }}-keycloak-controller-enamsse-admin
```
Comment (kornys) [inline templates/install/ansible/roles/standard_authservice/tasks/main.yml]: same typo

File: templates/install/ansible/roles/standard_authservice/tasks/main.yml [hunk 7]
```diff
@@ -12,8 +12,12 @@
...
+- name: Label clusterrolebindings {{ namespace }}-keycloak-controller-enamsse-admin
```
Comment (kornys) [inline templates/install/ansible/roles/standard_authservice/tasks/main.yml]: same typo

File: templates/install/ansible/roles/standard_authservice/tasks/main.yml [hunk 8]
```diff
@@ -12,8 +12,12 @@
...
+  shell: oc label --overwrite -n {{ namespace }} clusterrolebindings {{ namespace }}-keycloak-controller-enamsse-admin app=enmasse
```
Comment (kornys) [inline templates/install/ansible/roles/standard_authservice/tasks/main.yml]: same typo

File: templates/install/ansible/roles/address_controller_multitenant/tasks/main.yml [hunk 9]
```diff
@@ -6,8 +6,14 @@
 - name: Create cluster wide roles used by enmasse-admin service account
   shell: oc apply -n {{ namespace }} -f {{ playbook_dir }}/resources/cluster-roles/address-space-controller.yaml
 - name: Grant address space controller cluster role privileges to service account
-  shell: oc adm policy add-cluster-role-to-user enmasse.io:address-space-controller -n {{ namespace }} system:serviceaccount:{{ namespace }}:enmasse-admin
+  shell: oc adm policy add-cluster-role-to-user enmasse.io:address-space-controller -n {{ namespace }} system:serviceaccount:{{ namespace }}:enmasse-admin --rolebinding-name={{ namespace }}-address-space-controller-enamsse-admin
```
Comment (kornys) [inline templates/install/ansible/roles/address_controller_multitenant/tasks/main.yml]: typo enamsse should be enmasse

**Comment by dmvolod** [comment]
@kornys , sorry, fixed typos. Thanks.

**Comment by lulf** [comment]
@dmvolod On one hand, yes it is nice to have the labels for all resources, but having to run one extra command to add the labels etc. seems perhaps to be a potential source of bugs to me from a maintenance perspective. I.e. its really easy to 'forget' to label something. 

What is the use case for labelling all the 'cluster wide' resources? You wouldn't normally deploy EnMasse in the same namespace as other components, so if you want to delete everything, you should be able to do that using `oc delete all --all` in the enmasse namespace.

An alternative that we should think about for the playbooks is to use the kubernetes/openshift module instead of the cli directly, which would perhaps make the labeling as part of the create.

**Comment by dmvolod** [comment]
@lulf , yes you are right and I agree with you. If all are fine and users/admins will create enamsse resource in correct namespace it's a not problem with labelling, but i someone will wrong, it will be not able to delete correct resources without labels.
From 'cluster wide' resources point of view, if we will install and de-install enmasse again, OCP will keep this resources with several names and nobody care about remove these.
I understood that support labelling on active development cycle a little bit difficult in several places, such as ansible playbooks and deploy.sh script.
Do you mind to reduce number of 'oc label' invocations with a single one or remove changes from this PR?

**Comment by dmvolod** [comment]
@lulf , in advance to original message, please pay attention that 'oc delete all --all' is not enough to delete all resources inside namespace. It's necessary to add 'pvc,configmap,secrets,sa,roles' to delete all resources.

**Comment by lulf** [comment]
@dmvolod  If we can reduce the invocation to a single one as part of a 'label resources' step, maybe thats easier to manage I think.

**Comment by dmvolod** [comment]
Difficult to merge with changes

**Pull request closed without merging**